Sources/Just/Just.swift

```diff
@@ -998,13 +998,13 @@ public final class HTTP: NSObject, URLSessionDelegate, JustAdaptor {
       data: Data(),
       progressHandler: asyncProgressHandler)
     { result in
-      if let handler = asyncCompletionHandler {
-        handler(result)
-      }
       if isSynchronous {
         requestResult = result
         semaphore.signal()
       }
+      if let handler = asyncCompletionHandler {
+        handler(result)
```

Comment:
I switch those 2 blocks in case `asyncCompletionHandler` is too heavy, which may take more than `timeIntervalBetweenThread` seconds,  so that `semaphore.signal()` can be executed as soon as we get the result of request